codemp/rd-rend2/tr_weather.cpp

```diff
@@ -1143,6 +1143,11 @@ void RB_SurfaceWeather( srfWeather_t *surf )
 			{
 				chunkIndex  = (int(centerZoneOffsetX + numMinZonesX) + x + 1) % 3;
 				chunkIndex += (int(centerZoneOffsetY + numMinZonesY) + y + 1) % 3 * 3;
+
+				if (chunkIndex < 0) {
+					chunkIndex *= -1;
```

Comment:
Because the `%` operator uses _truncated division_ just flipping the sign won't lead to a continuous remainder. 
[![image](https://github.com/JACoders/OpenJK/assets/1207515/8a0114d0-c97d-4fe6-994d-2b696ffce9b5) ](https://en.wikipedia.org/wiki/Modulo)
With just flipping the sign, it causes clumping of the rain in the shader. 
![image](https://github.com/JACoders/OpenJK/assets/1207515/46befaed-9ba2-496c-ad05-592bd8d9b18b)

Reply:
Oh, yeah! You are right. I flew around and didn't spot that issue.

Reply:
I do not know if flipping the sign is actually the best fix all together either, but I could be wrong.